src/MockFactory.ts

```diff
@@ -64,111 +65,146 @@ const executeOperation = (
 
 interface Options {
   mocks?: ResolverMap
+  addTypename?: Boolean
+}
+
+interface MockOptions<TVariables> {
+  variables?: TVariables
+  mocks?: ResolverMap
+  addTypename?: Boolean
 }
 
 class MockFactory {
   private schema: GraphQLSchema
   private mocks: ResolverMap
+  private addTypename: Boolean
 
   private mockSchema(options: { mocks?: ResolverMap } = {}) {
     const clonedSchema = transformSchema(this.schema, [])
-    const mergedMocks = mergeResolvers(this.mocks, options.mocks || {})
+    const mergedMocks = mergeResolvers(this.mocks, options.mocks ?? {})
     addMockFunctionsToSchema({ schema: clonedSchema, mocks: mergedMocks })
     return clonedSchema
   }
 
-  constructor(schema: IntrospectionQuery, { mocks }: Options) {
-    this.schema = generateMockSchema(schema)
-    this.mocks = mocks || {}
+  private maybeAddTypenameToDocument = (
+    document: DocumentNode,
+    overrideAddTypename: Boolean | undefined,
+  ) => {
+    const addTypename = overrideAddTypename ?? this.addTypename
+    // ? parse(print(addTypenameToDocument(document)))
+    return addTypename ? addTypenameToDocument(document) : document
+  }
+
+  constructor(
+    schema: IntrospectionQuery,
+    { mocks, addTypename = false }: Options = {},
+  ) {
+    this.schema = buildMockSchema(schema)
+    this.mocks = mocks ?? {}
+    this.addTypename = addTypename
   }
 
-  mockFragment = <TData = any>(
+  mockFragment = <TData = any, TVariables = any>(
```

Comment:
Variables for a fragment is nice.

Reply:
Ya it would be except it doesn't work. Thanks for pointing this out so I can remove it for now 😢